src/test/javascript/spec/component/exam/manage/exams/exam-checklist.component.spec.ts

```diff
@@ -0,0 +1,215 @@
+                    { path: 'course-management/:courseId/exams/:examId/test-runs', component: DummyComponent },
+                    { path: 'course-management/:courseId/exams/:examId/students', component: DummyComponent },
+                ]),
+                HttpClientTestingModule,
+            ],
+            declarations: [
+                ExamChecklistComponent,
+                DummyComponent,
+                MockPipe(TranslatePipe),
+                MockPipe(ArtemisDatePipe),
+                MockComponent(AlertComponent),
+                MockComponent(AlertErrorComponent),
+                MockDirective(JhiTranslateDirective),
+                MockDirective(HasAnyAuthorityDirective),
+                ExamChecklistCheckComponent,
+                ExamChecklistExerciseGroupTableComponent,
+                ProgressBarComponent,
+                MockDirective(NgbTooltip),
+                MockComponent(FaIconComponent),
+            ],
+            providers: [
+                {
+                    provide: ActivatedRoute,
+                    useValue: {
+                        data: {
+                            subscribe: (fn: (value: Data) => void) =>
+                                fn({
+                                    exam,
+                                }),
+                        },
+                        snapshot: {},
+                    },
+                },
+                MockProvider(AccountService, {
+                    isAtLeastInstructorInCourse: () => true,
+                }),
+            ],
+            schemas: [],
+        })
+            .compileComponents()
+            .then(() => {
+                examChecklistComponentFixture = TestBed.createComponent(ExamChecklistComponent);
+                examDetailComponent = examChecklistComponentFixture.componentInstance;
+                exerciseGroupService = TestBed.inject(ExerciseGroupService);
+                findAllForExamStub = stub(exerciseGroupService, 'findAllForExam');
+                findAllForExamStub.returns(of(responseExerciseGroup));
+            });
+    });
+
+    beforeEach(function () {
+        // reset exam
+        exam.id = 1;
+        exam.title = 'Example Exam';
+        exam.numberOfRegisteredUsers = 3;
+        exam.maxPoints = 100;
+        exam.course = new Course();
+        exam.course.id = 1;
+        examDetailComponent.exam = exam;
+
+        examChecklist.numberOfGeneratedStudentExams = 1;
+    });
+
+    afterEach(function () {
+        sinon.restore();
+    });
+
+    describe('test checkTotalPointsMandatory', () => {
+        it('should set totalPointsMandatory to false', () => {
+            examDetailComponent.exam.exerciseGroups = exerciseGroupsExercisePointsEqual;
+            examDetailComponent.checkTotalPointsMandatory();
+            expect(examDetailComponent.totalPointsMandatory).to.be.equal(false);
```

Comment:
These three lines get repeated so maybe move them into a function or beforeEach.

Reply:
I refactored some of the code to reduce duplication!